tools/dotnet-linker/Steps/CollectAssembliesStep.cs

```diff
@@ -19,6 +19,10 @@ protected override void TryProcess ()
 			var getReferencedAssemblies = Configuration.Context.GetType ().GetMethod ("GetReferencedAssemblies");
 			var assemblies = (IEnumerable<AssemblyDefinition>) getReferencedAssemblies.Invoke (Configuration.Context, new object [0]);
 			Configuration.Assemblies.AddRange (assemblies);
+			foreach (var assembly in assemblies) {
+				if (!Configuration.AssembliesByName.TryAdd (assembly.Name.Name, assembly))
+					throw new InvalidOperationException ($"Multiple assemblies with the same name: {assembly.Name.Name}");
```

Comment:
Is this not something handled by the linker itself, i.e. before it reach this code

Reply:
You're right, somehow I missed that we can just use GetLoadedAssembly